core-processor/src/processing.rs

```diff
@@ -147,11 +147,49 @@ where
                 res.system_reservation_context,
                 ActorExecutionErrorReplyReason::Trap(reason),
             ),
-            DispatchResultKind::Success => process_success(Success, res),
+            DispatchResultKind::Success => {
+                if let (Some(current_prev), Some(current_res)) = (
```

Comment:
couldn't matching and extraction be hidden in debug_assert to avoid excess code in prod compile?